build/xunit.runsettings

```diff
@@ -22,18 +22,36 @@
     <DiagnosticMessages>true</DiagnosticMessages>
 
     <!-- Set this value to enable long-running (hung) test detection. When the runner is idle waiting for tests to finished, it will report that fact once the timeout has passed. Use a value of 0 to disable the feature, or a positive integer value to enable the feature (time in seconds). -->
-    <LongRunningTestSeconds>120</LongRunningTestSeconds>
+    <LongRunningTestSeconds>90</LongRunningTestSeconds>
 
     <!-- Set this to override the maximum number of threads to be used when parallelizing tests within this assembly. Use a value of 0 or "default" to indicate that you would like the default behavior; use a value of -1 or "unlimited" to indicate that you do not wish to limit the number of threads used for parallelization. -->
-    <MaxParallelThreads>4</MaxParallelThreads>
+    <MaxParallelThreads>0</MaxParallelThreads>
 
     <!-- Set this to true if this assembly is willing to participate in parallelization with other assemblies. Test runners can use this information to automatically enable parallelization across assemblies if all the assemblies agree to it. -->
-    <ParallelizeAssembly>false</ParallelizeAssembly>
+    <ParallelizeAssembly>true</ParallelizeAssembly>
 
     <!-- Set this to true if the assembly is willing to run tests inside this assembly in parallel against each other. Tests in the same test collection will be run sequentially against each other, but tests in different test collections will be run in parallel against each other. Set this to false to disable all parallelization within this test assembly. -->
-    <ParallelizeTestCollections>false</ParallelizeTestCollections>
+    <ParallelizeTestCollections>true</ParallelizeTestCollections>
 
     <!-- Set this to true to use shadow copying when running tests in separate App Domains; set to false to run tests without shadow copying. When running tests without App Domains, this value is ignored. -->
     <ShadowCopy>false</ShadowCopy>
   </xUnit>
+
+  <LoggerRunSettings>
+    <Loggers>
+      <Logger friendlyName="blame" enabled="True" />
+    </Loggers>
+  </LoggerRunSettings>
+
+  <DataCollectionRunSettings>
+    <DataCollectors>
+      <!-- Enables blame -->
+      <DataCollector friendlyName="blame" enabled="True">
+        <Configuration>
+          <!-- Enables hang dump or testhost and its child processes when a test hangs for more than 5 minutes. Dump type "Full", "Mini" or "None" (just kill the processes). -->
+          <CollectDumpOnTestSessionHang TestTimeout="5min" HangDumpType="Mini" />
```

Comment:
Where do dumps get written?  Does our ci yaml save them as build artifacts?

Reply:
Yeah sorry, I wasn't sure if anyone cared.  The blame collector in the VS test platform always writes the dumps to `%TestRunDirectory%\<GUID>\<processname>_<processid>_<timestamp>_hangdump.dmp`

For example:

`D:\a\_work\_temp\aa55eaf4-acab-41bb-b98d-a40ade186c60\testhost_9960_20241023T034051_hangdump.dmp`

It also includes the memory dumps as test attachments and then the `DotNetCoreCLI` Azure DevOps tasks publishes the results including the attachments.

https://dev.azure.com/dnceng-public/public/_build/results?buildId=850774&view=logs&j=5f8fe092-e504-55d5-bea6-4c19fb56d81e&t=32933acc-9193-5e17-f0b5-79ab664378f0&l=247

So then if you click on the [test run](https://dev.azure.com/dnceng-public/public/_build/results?buildId=850774&view=ms.vss-test-web.build-test-results-tab&runId=22091912&paneView=debug):

![image](https://github.com/user-attachments/assets/b7a3907b-7326-4ae2-98b4-de006cd43589)

You can [download the attachments](https://dev.azure.com/dnceng-public/public/_build/results?buildId=850774&view=ms.vss-test-web.build-test-results-tab&runId=22091912&paneView=attachments):

![image](https://github.com/user-attachments/assets/fbdb32cb-5fef-4cfe-8e91-74976893a6ba)

I tried loading the memory dump but I couldn't get symbols to load, we might need to start uploading those artifacts on failure.  I'm going to follow up on that.

For now, the collector also prints out this awesome info:

```
The active test run was aborted. Reason: Test host process crashed
Data collector 'Blame' message: The specified inactivity time of 5 minutes has elapsed. Collecting hang dumps from testhost and its child processes.
Data collector 'Blame' message: Dumping 9960 - testhost.
  
The active Test Run was aborted because the host process exited unexpectedly. Please inspect the call stack above, if available, to get more information about where the exception originated from.
The test running when the crash occurred: 
 
 NuGet.XPlat.FuncTest.ListPackageTests.ListPackage_WithPrivateHttpSourceCredentialServiceIsInvokedAsNeeded_Succeeds
```

That test specifically is setting a static singleton, `HttpHandlerResourceV3.CredentialService`, which is my guess as to the issue.